Added query endpoint that takes in arguments and keeps ip as a global…

mahi-pas opened the pull request:
… variable. It only gets edited when a new variable is submitted and persists on other queries.

Signed-off-by: Mahi <mahipasarkar@gmail.com>

mahi-pas committed:
Added query endpoint that takes in arguments and keeps ip as a global variable. It only gets edited when a new variable is submitted and persists on other queries.

Signed-off-by: Mahi <mahipasarkar@gmail.com>

joewxboy reviewed (changes requested):
Great start, thanks!

Let's add one additional feature ... inspect the contents of the "ip" variable and ensure that it is in the form of a dotted quad: "xxx.xxx.xxx.xxx" where "xxx" is a number between 0 and 255.  If it isn't, then set it to null.

joewxboy commented on qr_code_generator/query.py:
```diff
@@ -0,0 +1,29 @@
+    parameters = request.args.copy()
+    if 'ip' in parameters:
+        ip = parameters.get('ip')
+
```
Here I would expect you to have a conditional using .fullmatch() and a regular expression matching an IP address.  See https://www.golinuxcloud.com/python-pattern-matching/

mahi-pas replied:
I chose the ipaddress class because it would be guaranteed to be correct. If you would like, I can change it. Should I revise it and use pattern matching?

mahi-pas committed:
Added filtering so IP Address must be correct format

ip address also now stores as type ipaddress from Python's ipaddress class

mahi-pas committed:
Revert "Added filtering so IP Address must be correct format"

This reverts commit 1c03a574b027dbeabc7caef908136d7037f42f5d.

mahi-pas committed:
Added IP address filtering

Only valid IP addresses will be stored
IP addresses are now stored by ipaddress class instead of string

Signed-off-by: Mahi pasarkarmahi@gmail.com

mahi-pas committed:
Merge remote-tracking branch 'upstream/main' into Issue#6-query-parameters

mahi-pas commented:
I have completed the change you asked for

joewxboy commented:
@mahi-pas Thanks, the code looks good.  According to the record above ^^^ three of your commits failed due to missing sign-off.  Are you able to go back and sign those, or should we just close this PR and re-submit a new one with the sign-offs?

mahi-pas commented:
> @mahi-pas Thanks, the code looks good. According to the record above ^^^ three of your commits failed due to missing sign-off. Are you able to go back and sign those, or should we just close this PR and re-submit a new one with the sign-offs?

I'll try a new pull request

mahi-pas commented:
It's in a new pull request: https://github.com/open-horizon-services/service-qr-code-generator/pull/10
